.github/workflows/deploy-opentofu.yml

```diff
@@ -0,0 +1,80 @@
+name: Deploy OpenTofu
+
+on:
+  push:
+    paths:
+      - 'web/deploy/terraform/**'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4.2.2
+
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-region: ${{ inputs.aws_region }}
+          role-to-assume: arn:aws:iam::${{ secrets.AWS_ACCOUNT_ID }}:role/github-actions-role-shared
+
+      - name: Setup OpenTofu
+        uses: opentofu/setup-opentofu@v1
+
+      - name: Check for changes in reusable modules
+        id: changed-tofu-modules
+        uses: tj-actions/changed-files@v45
+        with:
+          files: |
+            web/deploy/terraform/modules/**
+
+      - name: Check for changes in shared deployment
+        id: changed-tofu-shared
+        uses: tj-actions/changed-files@v45
+        with:
+          files: |
+            web/deploy/terraform/shared/**
+
+      - name: Check for changes in staging deployment
+        id: changed-tofu-staging
+        uses: tj-actions/changed-files@v45
+        with:
+          files: |
+            web/deploy/terraform/staging/**
+
+      - name: Check for changes in production deployment
+        id: changed-tofu-production
+        uses: tj-actions/changed-files@v45
+        with:
+          files: |
+            web/deploy/terraform/production/**
```

Comment:
I see where you're coming from, but we don't really need to check the files for changes. That's exactly what Terraform/OpenTofu will do and it's one of the benefits of having an Infrastructure-as-Code approach. If the configuration in the files matches the infrastructure, nothing will be created, destroyed or updated.

Reply:
Addressed